lib/web/apps.go

```diff
@@ -407,3 +423,28 @@ func (h *Handler) proxyDNSNames() (dnsNames []string) {
 	}
 	return dnsNames
 }
+
+// appServerOrSPPageFromAppServerPage converts a ResourcePage containing AppServers to a ResourcePage containing AppServerOrSAMLIdPServiceProviders.
+func appServerOrSPPageFromAppServerPage(appServerPage apiclient.ResourcePage[types.AppServer]) apiclient.ResourcePage[types.AppServerOrSAMLIdPServiceProvider] {
+	var resources []types.AppServerOrSAMLIdPServiceProvider
```

Comment:
if we already know the exact size of the slice we need, it's better to make with the known size: https://stackoverflow.com/a/45423692 (this is talking specifically of a 3rd parameter called "cap", but the cap is the len if no cap is provided)

```suggestion
	var resources []types.AppServerOrSAMLIdPServiceProvider
	resource := make([]types.AppServerOrSAMLIdPServiceProvider, len(appServerPage))
```